### DISCUSSION TITLE
Unable to upload blob file in react-native app

### DISCUSSION THREAD

**Discussion by ethankong113** [supabase/supabase#9599] [Questions]
I'm trying to upload an image / video as a blob, but I keep getting this error `error:  {"error": "Internal Server Error", "message": "Cannot read properties of undefined (reading 'fields')", "statusCode": 500}`

Here's my sample code

```
const testUrl = "https://cdn.pixabay.com/photo/2017/11/22/18/49/shiba-2971192_960_720.jpg";
const resp = await fetch(testUrl);
const blob = await resp.blob();
const { data, error } = await supabase.storage
    .from("videos")
    .upload("test_video",  blob);
```

If I upload a string instead, it works fine.. so it shouldn't be a setup issue.

Setup:
- react-native with expo

**Comment by ethankong113** [comment]
Found this line in the doc
```
For React Native, using either Blob, File or FormData does not work as intended. Upload file using ArrayBuffer from base64 file data instead, see example below.
```

looks like base64 is the solution

**Comment by react-ghost** [comment]
I'm confused because i've seen a few videos showing that this is possible.

**Reply by ethankong113** [reply]
Not sure. But I got the above statement from the supabase doc. Plus it doesn't work for my build. It works fine if you convert the blob to ArrayBuffer first.
